mlir/lib/Dialect/Bufferization/IR/BufferizationOps.cpp

```diff
@@ -598,6 +622,51 @@ OpOperand &MaterializeInDestinationOp::getSourceOperand() {
   return getOperation()->getOpOperand(0) /*source*/;
 }
 
+LogicalResult MaterializeInDestinationOp::verify() {
+  if (!isa<TensorType, BaseMemRefType>(getDest().getType()))
+    return emitOpError("'dest' must be a tensor or a memref");
```

Comment:
Nit: no invalid test for this error. A vector could be used to test this, right?